wallet/src/repository/wallet/mod.rs

```diff
@@ -1321,22 +1440,11 @@ where
         let keychain = constants::EXTERNAL_KEYCHAIN;
         let account = state.account;
         let index = state.next_external_index;
-        let parent_key = &state.keychains[keychain as usize];
+        let parent_key = state.keychains[keychain as usize].clone();
 
         let (address, next_index) =
-            self.gen_address(label, parent_key, account, keychain, index)?;
-
-        let path = model::Path {
-            account,
-            keychain,
-            index,
-        }
-        .to_string();
+            self.gen_address(label, &parent_key, account, keychain, index, true)?;
```

Comment:
Deadlock? Even though the code is the same as before :thinking: 

```suggestion
            self._gen_address(label, &parent_key, account, keychain, index, true)?;
```